core/src/dpdk/stats.rs

```diff
@@ -117,3 +118,55 @@ impl PortStats {
         Ok(values)
     }
 }
+
+/// Mempool stats collector.
+pub struct MempoolStats {
+    raw: NonNull<ffi::rte_mempool>,
+}
+
+impl MempoolStats {
+    /// Builds a collector from the port.
+    pub fn build(mempool: &Mempool) -> Self {
+        MempoolStats {
+            raw: unsafe {
+                NonNull::new_unchecked(
+                    mempool.raw() as *const ffi::rte_mempool as *mut ffi::rte_mempool
+                )
+            },
+        }
+    }
+
+    fn raw(&self) -> &ffi::rte_mempool {
+        unsafe { self.raw.as_ref() }
+    }
+
+    /// Returns the name of the `Mempool`.
+    fn name(&self) -> &str {
+        self.raw().name[..].as_str()
+    }
+
+    /// Returns a gauge.
+    fn new_gauge(&self, name: &'static str, value: i64) -> (Key, Measurement) {
+        (
+            Key::from_name_and_labels(
+                name,
+                labels!(
+                    "pool" => self.name().to_string(),
+                ),
+            ),
+            Measurement::Gauge(value),
+        )
+    }
+
+    /// Collects the mempool stats.
+    pub fn collect(&self) -> Vec<(Key, Measurement)> {
+        let used = unsafe { ffi::rte_mempool_in_use_count(self.raw()) as i64 };
```

Comment:
with the unsafe call in here, should we do anything around wrapping this as a result w/ a possible error?

Reply:
doc didn't say this call can fail. should be safe.